### PR TITLE
Example for creating a new Ethernet connection profile

### PR DESCRIPTION
**Pull request by bernhardkaindl**

### CODE REVIEW

**Comment by bernhardkaindl** [comment]
@igo95862 This is also updated for `mypy --struct` and `flake8`.

I'd add the asyncio variant after the blocking variant passed review.

File: examples/block/add-eth-connection.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    functools.partial(pprint.pprint, sort_dicts=False)(profile)
+
+
+if __name__ == "__main__":
+    p = ArgumentParser(description="Optional arguments have example values:")
+    conn_id = "MyConnectionExample"
+    p.add_argument("-c", dest="conn_id", default=conn_id, help="Connection Id")
+    p.add_argument("-i", dest="interface_name", default="", help="ethX device")
+    p.add_argument("-4", dest="ip4", default="10.9.9.99/24", help="IP4/prefix")
+    p.add_argument("-g", dest="gw", default="10.9.9.1/40000", help="gw/metric")
```
Comment (igo95862) [inline examples/block/add-eth-connection.py:95]: What do you think about using documentation IP ranges?

https://www.rfc-editor.org/rfc/rfc5737

For example, 192.0.2.0/24 .

**Review by igo95862 — CHANGES_REQUESTED** [review]

File: examples/block/add-eth-connection.py (hunk 1 above)
Comment (bernhardkaindl) [reply examples/block/add-eth-connection.py]: Cool! Pushed with 192.0.2.0/24 !

Comment (bernhardkaindl) [reply examples/block/add-eth-connection.py]: I will add the async version next week when nothing else comes up before that!

**Review by igo95862 — APPROVED** [review]

**Review by igo95862 — CHANGES_REQUESTED** [review]
Would you mind rebasing against master so that commits e77fdf82c0ab4a3b2c798bdabbf05e0f573bcac0 and 8ba31e314bd6c91cd82a56d6be6284d91af70ec8 don't get double merged?

**Comment by bernhardkaindl** [comment]
> Would you mind rebasing against master so that commits [e77fdf8](https://github.com/python-sdbus/python-sdbus-networkmanager/commit/e77fdf82c0ab4a3b2c798bdabbf05e0f573bcac0) and [8ba31e3](https://github.com/python-sdbus/python-sdbus-networkmanager/commit/8ba31e314bd6c91cd82a56d6be6284d91af70ec8) don't get double merged?

Done: Rebased #8 on new master and #4 on the rebased #8.

If you merge the rebased #8 using the "Rebase and merge" method of github (it's the 3rd of the 3 choices on how github can merge), you would not create a merge commit for #8 and it would be applied on master with no change, and then you can merge this #4 in one go without a need to rebase this one again on the otherwise created merge commit:

Ready to merge #8 by "Rebase and merge strategy, without creating a merge commit",
and then #4 after this merge off #8.

**Comment by bernhardkaindl** [comment]
And of course I always accept github default of allowing project maintainers write access to the branch of the PR, so you can make any changes you want to the PR, including force-pushes with rebases.

And I can force-push the removal the commit for #8 from this PR, then you can merge #8 also with merge commit and this can be merged after it without rebase as the patch-context does not conflict.

**Comment by bernhardkaindl** [comment]
@igo95862: Finished the rebase: It now only has a single commit on current master:

You can merge now, and it wil work when `get_connections_by_id()` like from #8 is merged too.

**Pull request merged**